lib/message/context.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * The context is an abstraction we use to add domain separation to OpenPGP signatures.
+ * It's implemented by adding notation data to signatures, marked as critical, so that
+ * the resulting signature can only be verified by a verifier who expects the context to be present.
+ */
+import type { SignaturePacket, RawNotation, PartialConfig } from '../openpgp';
+
+// Different contexts will affect the notation's value, not the name
+const CONTEXT_NOTATION_NAME = 'context@proton.ch';
+
+/**
+ * Translate a `context` string into an OpenPGP notation object, which can be signed as part of the message
+ * @param context - context identifier
+ */
+export const getNotationFromContext = (context: string): RawNotation => ({
```

Comment:
Super nitpicky but this should be called `getNotationForContext` or `createNotationForContext`; `getNotationFromContext` would imply that you're taking the notation out of the context parameter somehow :)